### DISCUSSION TITLE
Zidoo fork merging upstream?

### DISCUSSION THREAD

**Discussion by Sledgehamma** [jellyfin/jellyfin-androidtv#2759] [General]
There is a fork of this Android version by Andy2244 (https://github.com/Andy2244/jellyfin-androidtv-zidoo) which incorporates the API of Zidoo (hardware media player based upon Realtek SoC) to run its native player. 
As the development of the fork seems to have sadly stopped I had one question in mind and before trying to do so I'd like to ask you:
Would you merge the specific Zidoo API upstream if it would be selectable as a player option the user can choose from. I was thinking adding it under "Preferred media player"as another option like "Zidoo player". 
This integration would be far better than just using the Zidoo player as an external player (playback status, playlist, loading screen etc.).

**Comment by nielsvanvelzen** [comment]
The author of the code should upstream it. Unfortunately Andy decided to stop upstreaming his changes because he didn't want to address review feedback, thus it seems unlikely the changes from the fork will be added to the main app. If someone else writes their own implementation we can of course always see if the changes fit the app.

For adding zidoo player support, we can extend our external player code to send/receive data when the external player is the zidoo option but it would still be using the "external player" option, we won't be adding it as a separate player type.

**Comment by Sledgehamma** [comment]
Alright, thanks for letting me know :)
I will have a look how the Zidoo API can be integrated into the external player option. Never did it and not much spare time, so it might take awhile, though^^
Nice weekend!
